packages/amplify-graphql-auth-transformer/src/__tests__/multi-auth.test.ts

```diff
@@ -620,7 +620,11 @@ describe('iam checks', () => {
     expect(out).toBeDefined();
     const createResolver = out.resolvers['Mutation.createPost.auth.1.req.vtl'];
     expect(createResolver).toContain(
-      `#if( ($ctx.identity.userArn == $ctx.stash.authRole) || ($ctx.identity.cognitoIdentityPoolId == \"${identityPoolId}\" && $ctx.identity.cognitoIdentityAuthType == \"authenticated\") )`,
+      `#if( $util.authType() == "IAM Authorization" )
+  #if( $ctx.identity.userArn == $ctx.stash.authRole )
```

Comment:
In addition to the authRole, we should also check whether the user is authenticated like before.

Reply:
that path is no longer hit in `iamExpression` method, and instead of setting isAuthorized we're setting allowed fields. lmk if you want to discuss more